core/jobs/playtime_entries.py

```diff
@@ -218,3 +238,56 @@ def _fetch_report_records(self, start: datetime, until: datetime) -> Query:
                 PlaytimeSummary.identifier_id,
             )
         )
+
+    @staticmethod
+    def _isbn_for_identifier(
+        identifier: Optional[Identifier],
+        /,
+        *,
+        default_value: str = "",
+    ) -> str:
+        """Find the strongest ISBN match for the given identifier.
+
+        :param identifier: The identifier to match.
+        :param default_value: The default value to return if the identifier is missing or a match is not found.
+        """
+        if identifier is None:
+            return default_value
+
+        if _is_usable_isbn_identifier(identifier):
+            return cast(str, identifier.identifier)
+
+        # If our identifier is not an ISBN itself, we'll use our Recursive Equivalency
+        # mechanism to find the next best one that is, if available.
+        db = Session.object_session(identifier)
+        eq_subquery = db.query(RecursiveEquivalencyCache.identifier_id).filter(
+            RecursiveEquivalencyCache.parent_identifier_id == identifier.id
+        )
+        equivalent_identifiers = (
+            db.query(Identifier)
+            .filter(Identifier.id.in_(eq_subquery))
+            .filter(Identifier.type == Identifier.ISBN)
+        )
+
+        isbn = next(
+            map(
+                lambda id_: id_.identifier,
+                filter(
+                    lambda id_: _is_usable_isbn_identifier(id_), equivalent_identifiers
```

Comment:
I guess not. I meant to handle an edge case, since it seems possible to have an empty or null value for the identifier. It does seem unlikely, so I'm comfortable removing the checks. I think I'll just remove the check everywhere and count on the `identifier` string always being available for ISBN identifiers.